pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -779,7 +832,128 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1.Mac
 		return &clustererror.RequeueAfterError{RequeueAfter: time.Second * 5}
 	}
 
-	// node is now available again
+	// node is now available again - restore annotations and labels
+	if _, annotationsBackupExists := machine.Annotations[nodeAnnotationsBackupAnnotation]; annotationsBackupExists {
+		return a.restoreAnnotationsAndLabels(ctx, node, machine)
+	}
+
+	//remediation is done
 	log.Printf("Node %s is available, remediation of Machine %s complete", node.Name, machine.Name)
 	return a.deleteRemediationAnnotations(ctx, machine)
 }
+
+// storeAnnotationsAndLabels copies node's annotations and labels and stores them on machine's annotations
+func (a *Actuator) storeAnnotationsAndLabels(ctx context.Context, node *corev1.Node, machine *machinev1.Machine) error {
+	marshaledAnnotations, err := marshal(node.Annotations)
+	if err != nil {
+		log.Printf("Failed to marshal node %s annotations associated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return nil
+	}
+
+	marshaledLabels, err := marshal(node.Labels)
+	if err != nil {
+		log.Printf("Failed to marshal node %s labels associated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return nil
+	}
+
+	machine.Annotations[nodeLabelsBackupAnnotation] = marshaledLabels
+	machine.Annotations[nodeAnnotationsBackupAnnotation] = marshaledAnnotations
+
+	err = a.client.Update(ctx, machine)
+	if err != nil {
+		log.Printf("Failed to update machine with node's annotations and labels %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// restoreAnnotationsAndLabels copies annotations and labels stored in machine's annotation to the node
+// and removes the annotations used to store that data
+func (a *Actuator) restoreAnnotationsAndLabels(ctx context.Context, node *corev1.Node, machine *machinev1.Machine) error {
+	if len(machine.Annotations) == 0 {
+		return &clustererror.RequeueAfterError{}
```

Comment:
Basically we will never reach this line since we enter this func only after checking that annotation exists (line #836).

My thought was to protect this function without assuming anything about the given Machine state, and if there are no annotations, we want to proceed to next remediation step, and since there's no change to the Machine object in this case, we need a requeue to be triggered again.

It might be redundant, and maybe we should remove that. 
WDYT?

Reply:
Seems reasonable. It doesn't hurt to bail out and try reconciliation again if something unexpected happens. It may help to just add a comment here, and perhaps even a log statement, in case this ever does get called and causes confusion.